src/components/flatSelect/FlatSelectReducers.ts

```diff
@@ -0,0 +1,54 @@
+import { IReduxAction } from '../../utils/ReduxUtils';
+import * as _ from 'underscore';
+import { FlatSelectActions, IFlatSelectActionPayload } from './FlatSelectActions';
+import { IFlatSelectOptionProps } from './FlatSelectOption';
+
+export interface IFlatSelectState {
+  id: string;
+  selectedOption?: IFlatSelectOptionProps;
+}
+
+export const flatSelectInitialState: IFlatSelectState = { id: undefined, selectedOption: undefined };
+export const flatSelectsInitialState: IFlatSelectState[] = [];
+
+export const flatSelectReducer = (state: IFlatSelectState = flatSelectInitialState,
+  action: (IReduxAction<IFlatSelectActionPayload>)): IFlatSelectState => {
+  switch (action.type) {
+    case FlatSelectActions.select:
+      if (state.id !== action.payload.id) {
+        return state;
+      }
+
+      return {
+        id: state.id,
+        selectedOption: action.payload.selectedOption,
+      };
+    case FlatSelectActions.add:
+      return _.extend({}, action.payload);
```

Comment:
you don't need underscore and could do `return {...action.payload};`

Reply:
I got a error ts caused by IReduxAction. I will keep the _.extend